- adds change notifications payload types in v1.0 (#8811)
microsoftgraph/microsoft-graph-docs: api-reference/v1.0/resources/changenotification.md

---
title: "changeNotification resource type"
description: "Represents the change notification sent to the subscriber."
localization_priority: Normal
author: "baywet"
doc_type: resourcePageType
ms.prod: "non-product-specific"
---

# changeNotification resource type

Namespace: microsoft.graph

Represents the notification sent to the subscriber.

For details, see [Use the Microsoft Graph API to get change notifications](webhooks.md).

## Methods

None.

## Properties

| Property | Type | Description |
|:---------|:-----|:------------|

> | changeType | string | Indicates the type of change that will raised the change notification. The supported values are: `created`, `updated`, `deleted`. Required. |

Reviewer: Indicates the type of change that will raise the change notification.

Author: ```suggestion
| changeType | string | Indicates the type of change that will raise the change notification. The supported values are: `created`, `updated`, `deleted`. Required. |
```

> | clientState | string | Value of the **clientState** property sent specified in the subscription request (if any). The maximum length is 255 characters. The client can check that the change notification came from the service by comparing the value of the **clientState** property sent with the subscription with the value of the **clientState** property received with each change notification. Optional. |

Reviewer: Value of the **clientState** property sent in the subscription request (if any). The maximum length is 255 characters. The client can check whether the change notification came from the service by comparing the values of the **clientState** property. The value of the **clientState** property sent with the subscription is compared with the value of the **clientState** property received with each change notification.

Author: ```suggestion
| clientState | string | Value of the **clientState** property sent in the subscription request (if any). The maximum length is 255 characters. The client can check whether the change notification came from the service by comparing the values of the **clientState** property. The value of the **clientState** property sent with the subscription is compared with the value of the **clientState** property received with each change notification. Optional. |
```

| id | string | Unique ID for the notification. Optional. |
| resource | string | The URI of the resource that emitted the change notification relative to `https://graph.microsoft.com`. Required. |

> | resourceData | [microsoft.graph.resourceData](resourcedata.md) | The content of this property depends on the type of resource being subscribed to. Required. |

Reviewer: Is the user going to know where to find these types?

Author: yes this really depends on what value they put in the `resource` property when creating the subscription.

| sequenceNumber | int | A number in sequence for a notification, to help the client app identify if notifications are in sequence or if a notification is missing. Optional. |
| subscriptionExpirationDateTime | [dateTime](https://tools.ietf.org/html/rfc3339) | The expiration time for the subscription. Required. |
| subscriptionId | string | The unique identifier of the subscription that generated the notification. |

> | tenantId | guid | The unique identified of the tenant from which the change notification originated. |

Reviewer: The unique identifier of the tenant from which the change notification originated.

Author: ```suggestion
| tenantId | guid | The unique identifier of the tenant from which the change notification originated. |
```
